### ISSUE TITLE
Chrome 114 now has the same performance as Thorium

### ISSUE THREAD

**Issue by People-11** [Alex313031/thorium#167]
Since it uses Maglev compiler on it, it has huge performance imporvement. Do Thorium still have confidence to beat it?

![image](https://github.com/Alex313031/thorium/assets/41558215/73be7aa2-6609-403b-b073-6a170491d763)
![image](https://github.com/Alex313031/thorium/assets/41558215/cab43c0e-14a8-4741-955d-f53d29d373f1)
(Left Thorium Right Chrome 114)

**Comment by gz83** [comment]
Comparing different versions is not accurate and reliable.

Thorium constantly adjusts the optimization parameters according to the actual situation.

Also, not as many optimizations as possible is better.

@People-11

**Comment by Alex313031** [comment]
@gz83 This is true, however, the versions are close enough that I think it still shows that Chromium has caught up in terms of performance, which is a good thing, but I also always want to beat it lolol.

@People-11 Are you using thorium 112 or 113 for these screenshots.

**Comment by People-11** [comment]
@Alex313031 I am using Throium 112.0.5615.166 AVX2 since 113 is not yet available for AVX2.

**Comment by Alex313031** [comment]
@People-11 Ok i brought in some optimizations from @RobRich999 that should improve performance

**Comment by rohitsaha29** [comment]
> @People-11 Ok i brought in some optimizations from @RobRich999 that should improve performance
@Alex313031 did you push these optimizations to AVX2 yet? also, any idea when we might get 113 for AVX2?

**Comment by gz83** [comment]
The optimization parameters will be passed into the BUILD.gn file of different architectures or instruction sets. In addition, I will compile the avx2 version for windows system in the near future.

@rohitsaha29

**Comment by rohitsaha29** [comment]
got it, so I should just wait on you to update the AVX2 download to the new optimizations/113?

@gz83

**Comment by gz83** [comment]
The new AVX2 version has been compiled, please wait patiently for Alex to release them

@rohitsaha29

**Comment by People-11** [comment]
Perofrmance test between Chrome 114 and latest Thorium 113 AVX
<img width="1280" alt="Snipaste_2023-06-14_02-42-21" src="https://github.com/Alex313031/thorium/assets/41558215/112b9501-4acf-496b-859c-013c1cc23d18">
<img width="1280" alt="Snipaste_2023-06-14_02-45-54" src="https://github.com/Alex313031/thorium/assets/41558215/03780a11-ca07-4649-a220-5f60bf666a2b">

**Comment by midzer** [comment]
@People-11 Hi and thanks for those screenshots! As you already wrote M114 introduces Maglev:

> It reduces overall CPU time to compile code while also saving battery life. Our measurements show Maglev has provided a 7.5 percent improvement on Jetstream and a 5 percent improvement in Speedometer.

via https://blog.chromium.org/2023/06/how-chrome-achieved-high-scores-on.html

So, in this case it is not fair to compare Thorium M113 with Chrom* 114.

Let's wait for M114 and compare again :)

**Comment by rohitsaha29** [comment]
Can’t wait for the newest versions (114 AVX2 & 114 MacOS) hopefully they’re soooon

**Comment by Aedon** [comment]
@People-11 avx2 114 out!

**Comment by Talent30** [comment]
What is the benchmark of avx2 114?

**Comment by midzer** [comment]
Miserably, Thorium M114 for MacOS ARM64 lacks behind Chrome M114 about ~10%. We don't know what Google is doing differently for building their browser right now.

Can't test MacOS x64.

Please post other OS builds in here if you like.

**Comment by RobRich999** [comment]
My dev build is three major revisions ahead, so that makes for a big YMMV on performance numbers. Of note I also only benchmark my builds on Linux, and even then it is mostly rather casual. I do not have much use for Windows.

Ideally we probably should be generating our own PGO profiles when changing the SIMD baseline. Function counts should be the same, but we are altering functions, so.... yeah. ?? It is just annoying and time consuming to do. Been there, done that back in the MSVC days of Chromium on Windows.

**Comment by People-11** [comment]
Both are 114.0.5735.134, Left is Chrome and Right is Thorium
<img width="1280" alt="Snipaste_2023-06-27_01-28-12" src="https://github.com/Alex313031/thorium/assets/41558215/b6eacdcc-fcad-4bb5-8e3e-79de264c2962">
<img width="1280" alt="Snipaste_2023-06-27_01-31-14" src="https://github.com/Alex313031/thorium/assets/41558215/93267d29-0236-499d-bae8-a43066ca2e26">

**Comment by dmig** [comment]
Guys providing these pics with insane numbers, describe your hardware.

Went curious and ran this test myself: on my i7-10xxx intel video it's **142/172** in speedtest (chrome/thorium).
The difference is even visually noticeable. While rendering in Chrome seems smoother (not faster, just smoother), video is noticeably sluggish.

**Comment by rohitsaha29** [comment]
@dmig Yeah with the newest Thorium version, I have noticed the same things (thumbnails or videos seems sluggish and buggy while scrolling or loading). Haven’t really tested chrome yet as I thought it would just fix over time. 113 thorium didn’t have these types of “issues” however. 

I am using stock flags besides enable dark mode, prominent dark mode and tab muting flags. same extensions on both. Win10,i7-13700k, 2080ti.

**Comment by People-11** [comment]
Thorium 119.0.6045.214 AVX2 (left) vs. Chrome 120.0.6099.200
<img width="1280" alt="Snipaste_2024-01-10_05-05-17" src="https://github.com/Alex313031/thorium/assets/41558215/5a1b798b-b1af-48c0-8462-2a712cd7102a">
Just for reference bcs the versions don't match.

**Comment by Alex313031** [comment]
@People-11 Thanks. It is not going to be super accurate, but this does show that me and/or @RobRich999 can work to eek out some more performance out of Chromium.

**Comment by user334** [comment]
MBP M1 @ macOS 13.6.3 test results:

Chrome Stable 120.0.6099.216 arm64
<img width="664" alt="image" src="https://github.com/Alex313031/thorium/assets/1550256/4d68173e-cd38-45a9-8f72-a40d4928d074">

Thorium 119.0.6045.214 arm64
<img width="691" alt="image" src="https://github.com/Alex313031/thorium/assets/1550256/e61c83b2-339e-4951-84d4-9391cde0fc90">

Just for the stats.

**Comment by People-11** [comment]
120.0.6099.235 AVX2
<img width="1280" alt="Snipaste_2024-01-25_11-52-20" src="https://github.com/Alex313031/thorium/assets/41558215/e271418d-00f7-4fa5-972b-599def99001f">

**Comment by People-11** [comment]
<img width="1276" alt="Snipaste_2024-02-21_03-34-37" src="https://github.com/Alex313031/thorium/assets/41558215/982707eb-ad3c-4cc3-b5f3-f311d96db56f">
Google Chrome 121.0.6167.185 vs. Thorium 121.0.6167.204

**Comment by gz83** [comment]
https://github.com/Alex313031/Thorium-Win/issues/168

It is quite similar to this issue. In addition, browser performance varies from person to person, so we do not need to spend too much time testing it.

@People-11

**Comment by midzer** [comment]
@People-11 There is nothing wrong with posting testing results.

Would be nice to have them on https://thorium.rocks/performance for a variety of devices, as written in https://github.com/Alex313031/Thorium-Win/issues/168#issuecomment-1954707532
